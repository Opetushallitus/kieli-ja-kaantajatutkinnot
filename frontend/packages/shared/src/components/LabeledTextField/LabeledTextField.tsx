import {
  CustomTextField,
  CustomTextFieldProps,
} from '../CustomTextField/CustomTextField';
import { Text } from '../Text/Text';

export type LabeledTextFieldProps = {
  id: string;
  label: string;
} & CustomTextFieldProps;

export const LabeledTextField = ({
  id,
  label,
  placeholder,
  error,
  ...rest
}: LabeledTextFieldProps) => {
  const errorStyles = error ? { color: 'error.main' } : {};

  return (
    <div className="rows">
      <label htmlFor={id}>
        <Text sx={errorStyles}>
          <b>{label}</b>
        </Text>
      </label>
      {placeholder && <Text sx={errorStyles}>{placeholder}</Text>}
      <CustomTextField id={id} error={error} {...rest} />
    </div>
  );
};
