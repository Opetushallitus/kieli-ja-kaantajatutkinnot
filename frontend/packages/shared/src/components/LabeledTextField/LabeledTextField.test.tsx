import renderer from 'react-test-renderer';

import { LabeledTextField } from './LabeledTextField';

describe('CustomTextField', () => {
  it('should render correctly with text', () => {
    const tree = renderer
      .create(
        <LabeledTextField id="test-id" label="test label" value="test value" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with error', () => {
    const tree = renderer
      .create(
        <LabeledTextField
          id="test-with-error"
          label="test label"
          placeholder="placeholder expected to appear between label and field"
          value="wrong test value"
          error={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
