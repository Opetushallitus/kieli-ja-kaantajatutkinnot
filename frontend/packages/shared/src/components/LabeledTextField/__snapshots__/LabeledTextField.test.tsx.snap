// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledTextField should render correctly with error 1`] = `
<div
  className="rows"
>
  <label
    htmlFor="test-with-error"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      <b>
        test label
      </b>
    </p>
  </label>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
  >
    placeholder expected to appear between label and field
  </p>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={true}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="test-with-error"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="wrong test value"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`LabeledTextField should render correctly with text 1`] = `
<div
  className="rows"
>
  <label
    htmlFor="test-id"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <b>
        test label
      </b>
    </p>
  </label>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="test-id"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="test value"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
